docs/HowToGuides.md

```diff
@@ -1,5 +1,78 @@
+
+Big PRs (i.e. PRs touching more than 100 lines) are hard to review. Using the _multi-step PR_ format allows the 
+review to be done incrementally.
+
+A multi-step PR consists a series of commits corresponding to a sequence of logical steps, 
+each step advancing the code base towards the end goal of the PR. The build must pass at each step. 
+
+                     
+> Build-breaking commits in the version history hinder the ability to use `git bisect` for locating bugs.
+
+[Here](https://github.com/se-edu/addressbook-level4/pull/237) is an example of a PR that is in multi-step format.
+The workflow of a multi-step PR is given below.
+
+1. Dev creates the PR branch in multi-step format.
+   
+   <details>
+   <summary>A typical workflow to achieve a multi-step commit sequence (yours may differ)</summary>
+   
+    
+   * Plan the PR as a sequence of logical steps.
+   * Implement each step, committing each step as a separate commit. Note that you might have to deviate from the 
+     original plan along the way, which is fine; continue to commit after each significant change.
+   * After the fix is complete, refactor the commits (i.e. reorder, squash, split, etc.) to achieve  
+     a commit sequence that reflects the logical steps of the fix you should have taken in retrospect.<br>
+     Remember to write [detailed commit messages](FormatsAndConventions.md#commit-message) for each commit.
+   </details>
+
+   
+1. When the PR is ready for review, dev posts a summary of commits using the 
+   [CanIHasReview tool](https://github.com/pyokagan/canihasreview/).
+    
+    <details>
+    <summary>How to use CanIHasReview</summary>
+    
+    1. Navigate to your PR. e.g. `https://github.com/se-edu/addressbook-level4/pull/237`.
+    1. Replace github.com in the PR URL with `canihasreview.herokuapp.com`. The resulting URL should be 
+       something like `https://canihasreview.herokuapp.com/se-edu/addressbook-level4/pull/237`.
+    1. Click `Submit new iteration` button. It will post a summary of the PR similar to 
+       [this example](https://github.com/se-edu/addressbook-level4/pull/209#issuecomment-270905049)
+    
+    </details>
+       
+1. The reviewer reviews the PR in an incremental fashion.
+
+    <details>
+    <summary>Insructions for the reviewer</summary>
+    
+    1. Review one commit at a time, starting with the earliest commit. 
+       [Here](https://github.com/se-edu/addressbook-level4/pull/209#pullrequestreview-15603608) is an example.
+    1. If the early commits require lot of changes, there's no need to review later commits until the early commits 
+       are updated as per review.
+    
+    </details>
+      
+1. After reach round of review, the dev _updates the existing commits_  (rather than add more commits)
+   according to the review comments. 
+   
+   <details>
+   <summary>A typical workflow for updating the PR in response to a review</summary>
+   
+   * Commit changes in separate commits.
```

Comment:
Maybe we should split up the points like this?

> * Updates to existing logical changes should be done by [modifying their corresponding commit](http://sethrobertson.github.io/GitPostProduction/gpp.html#analyze).
>
> * New logical changes should be introduced as new commits.
>
> * Sometimes, reviewers may recommend [splitting existing commits](http://sethrobertson.github.io/GitPostProduction/gpp.html#post-production) in order to make them more cohesive.
>
> * Commit messages should be updated with new findings from review discussions. For example, if the reviewer mentioned a possible new approach that was subsequently rejected due to a non-obvious reason, then the commit should be updated with this information.

The reason is because there are multiple ways to do them, and I think we should let this how-to be a "high level overview", For the nitty gritty of modifying a commit, we can then direct the reader to a page dedicated to that.